tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchCMISTests.java

```diff
@@ -123,6 +123,25 @@ public void objectIdQuery()
         searchQueryService.expectResultsFromQuery(query, user1, FILE_0_NAME);
     }
 
+    @TestRail (description = "Check folders can be selected using cmis:objectTypeId.", section = TestGroup.SEARCH, executionType = ExecutionType.REGRESSION)
+    @Test (groups = TestGroup.SEARCH)
+    public void objectTypeIdQuery()
+    {
+        SearchRequest query1 = req("cmis", "SELECT * FROM cmis:folder WHERE cmis:objectTypeId = 'cmis:folder'");
+        searchQueryService.expectResultsFromQuery(query1, user1,
+                // We expect the document library, ...
```

Comment:
Looks good. 

Do we really need the comments? Code is quite descriptive :)

Reply:
Comments removed.

Reply:
Aha! I think the "and the user home for the user performing the query" is quite useful, because it's not obvious why a username is a folder. Anyway - I think this is present somewhere else in the file, so no need to change it back.